server/src/bge/System.cpp

```diff
@@ -100,13 +263,13 @@ namespace bge {
                     glm::vec3 p0=pos.position+meshCol.collisionPoints[i];
                     // the t value that is returned is between 0 and 1; it is looking
                     // for a collision between p0+0*vel.velocity and p0+1*vel.velocity
-                    rayIntersection newInter=world->intersect(p0, vel.velocity, 1);
+                    rayIntersection newInter=world->intersect(p0, vel.velocity, meshCol.rayLength);
```

Comment:
I'm surprised to see this as a configurable value for each entity, since I can't imagine an object where we'd want to do movement-related collision detection past where it will move this tick (with meshCol.raylength=1, we basically check collisions up to where the farthest collision point would be at the next tick based on our velocity). What would we use this for? It doesn't hurt to have that ability, just curious about the thought process.

Reply:
i planned to merge players and bullet movement in one system (one while loop for bouncing rays). players would cast short rays to detect surroundings and bullets cast long ray. hence the variable maxT.

it turned out to be a mess. so I discarded that. 
now bullet has its own system (in bullets branch). so there's no need for varying maxT but I forgot to clean it up bruh...